### PR TITLE
[AOT Refactor] functional utils

### PR DESCRIPTION
**Pull request by jon-chuang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114643
* #114564
* #114563
* #114562
* #114561
* #114559
* #114558
* #114557
* #114556
* #114555
* #114554
* #114553
* __->__ #114552
* #114551
* #114550

---

Part _ of https://github.com/pytorch/pytorch/issues/114548

### CODE REVIEW

File: torch/_functorch/_aot_autograd/functional_utils.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
+import torch
+from torch import Tensor
+from torch._subclasses.functional_tensor import FunctionalTensor
+from torch.fx.experimental.symbolic_shapes import definitely_true, sym_eq
+from torch.multiprocessing.reductions import StorageWeakRef
+from torch.utils._python_dispatch import (
+    is_traceable_wrapper_subclass,
+    transform_subclass,
+)
```
Comment (bdhirsh) [inline torch/_functorch/_aot_autograd/functional_utils.py:9]: Can you add a brief description of which utils are meant for this file? (I think the word `functional` is pretty overloaded)

"This file contains functionalization-related utils used by AOTAutograd"

Comment (jon-chuang) [reply torch/_functorch/_aot_autograd/functional_utils.py:9]: Ok, I will add a description to every file

**Review by bdhirsh — APPROVED** [review]

**Pull request closed without merging**